### ISSUE TITLE
Feature request: Support `~` and `${HOME}` in shellcheck source

### ISSUE THREAD

**Issue by lamyergeier** [koalaman/shellcheck#2710]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1091, SC1090
- My shellcheck version (`shellcheck --version` or "online"):  0.9.0
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

Case 1

  ```sh
  #!/usr/bin/env bash
  
	  #shellcheck source=${HOME}/opt/bin/Framework
	  . Framework
  ```

Case 2

  ```sh
  #!/usr/bin/env bash
  
	  #shellcheck source=~/opt/bin/Framework
	  . Framework
  ```

#### Here's what shellcheck currently says:

Case 1: Not following: ${HOME}/opt/bin/Framework: openBinaryFile: does not exist (No such file or directory)
Case 2: ShellCheck can't follow non-constant source. Use a directive to specify location.

#### Here's what I wanted or expected to see:

To parse  `${HOME}` and `~`

**Comment by ELLIOTTCABLE** [comment]
I just ran into this as well:

```bash
rbenv_plugin_path="$HOME/.zsh/modules/ELLIOTTCABLE/rbenv.plugin.zsh/rbenv.plugin.zsh"
if [ -f "$rbenv_plugin_path" ]; then
   # shellcheck source=$HOME/.zsh/modules/ELLIOTTCABLE/rbenv.plugin.zsh/rbenv.plugin.zsh
   . "$rbenv_plugin_path"
```

**Comment by cwrau** [comment]
It would also be amazing if *all* environment variables would be followed, or maybe a parameter could be added with which a list could be set.

Because most of my environment variables are `XDG_*` vars 😅

**Comment by FriendlyNeighborhoodShane** [comment]
Agreed, it would be great to have _some_ amount of dynamism in the source directive.

For my use case, I have a script that chooses one of a bunch of config files at runtime, and uses variables and functions from it. According to recommendations, I created a static dummy config file and added that as a source directive to the main script, and it lints the flow of the main script great.
But it would be nice if I could run it on all of the various config files as well, by e.g. using:
```sh
# shellcheck source=${config_script}
```
with (the `--allow-source-vars` for obvious security considerations):
```sh
export config_script="./configs/conf-abc.sh";
shellcheck -ax --allow-source-vars=config_script ./main.sh;
```

---

RIght now this indirection can be (poorly) emulated using a proxy file as a poor man's `export`:
```sh
# shellcheck source=./configs/test-proxy.sh
```
```sh
for config_file in ./configs/conf-*.sh; do
    printf '. "%s";\n' "${config_file}" > ./configs/test-proxy.sh;
    shellcheck -ax ./main.sh;
done;
```
